Amqp.V1/Akka.Streams.Amqp.V1/AmqpSourceStage.cs

```diff
@@ -0,0 +1,68 @@
+﻿using Akka.Streams.Amqp.V1.Util;
+using Akka.Streams.Stage;
+using Akka.Streams.Supervision;
+using Amqp;
+using Amqp.Framing;
+using Amqp.Types;
+using System;
+
+namespace Akka.Streams.Amqp.V1
+{
+    public sealed class AmqpSourceStage<T> : GraphStage<SourceShape<T>>
+    {
+        public AmqpSourceStage(IAmqpSourceSettings<T> amqpSourceSettings)
+        {
+            AmqpSourceSettings = amqpSourceSettings;
+        }
+
+        public override SourceShape<T> Shape => new SourceShape<T>(Out);
+
+        public Outlet<T> Out { get; } = new Outlet<T>("AmqpSource.Out");
+        public IAmqpSourceSettings<T> AmqpSourceSettings { get; }
+
+        protected override GraphStageLogic CreateLogic(Attributes inheritedAttributes) => new AmqpSourceStageLogic(this, inheritedAttributes);
+
+        private class AmqpSourceStageLogic : GraphStageLogic
+        {
+            private readonly Outlet<T> outlet;
+            private readonly IAmqpSourceSettings<T> ampqSourceSettings;
+            private readonly ReceiverLink receiver;
+            private readonly Decider decider;
+
+            public AmqpSourceStageLogic(AmqpSourceStage<T> stage, Attributes attributes) : base(stage.Shape)
+            {
+                outlet = stage.Out;
+                ampqSourceSettings = stage.AmqpSourceSettings;
+                receiver = stage.AmqpSourceSettings.GetReceiverLink();
+                decider = attributes.GetDeciderOrDefault();
+
+                SetHandler(outlet, () => {
+                    var message = receiver.Receive();
+                    if (message != null)
```

Comment:
What if `message` is `null` ? You probably need to call `Receive` again ?!

Reply:
Can the receiver return `null` at all? If so, what it means?

Reply:
I don't know, I thought this OnPull was called by Akka every x times per second?